java/src/org/openqa/selenium/support/decorators/WebDriverDecorator.java

```diff
@@ -304,13 +304,20 @@ protected final <Z> Z createProxy(final Decorated<Z> decorated, Class<Z> clazz)
             || decoratedInterfaces.contains(method.getDeclaringClass())) {
           return method.invoke(decorated, args);
         }
-        if (originalInterfaces.contains(method.getDeclaringClass())) {
+        boolean isCompatible = originalInterfaces.stream()
```

Comment:
I understand the checks that are done to solve the issue. But it might be a good idea to name them separately instead of "isCompatible". It might help anyone reading the code understand what each does. Just a suggestion.

Reply:
@pujagani - I wasn't able to come up with a good descriptive name (which is short enough as well and not overly verbose) which explains very clearly what is happening. So I have instead added some comments that should help explain what is going on. Happy to rename the variable to something else if you can suggest something which you feel is more apt. We are essentially doing a compatibility check which is why I chose to name it as `isCompatible`

Reply:
@pujagani - I wasn't able to come up with a good descriptive name (which is short enough as well and not overly verbose) which explains very clearly what is happening. So I have instead added some comments that should help explain what is going on. Happy to rename the variable to something else if you can suggest something which you feel is more apt. We are essentially doing a compatibility check which is why I chose to name it as `isCompatible`

Reply:
I think the comments make the intent clear now. Thank you!